test/externalTests/gnosis.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+
+# ------------------------------------------------------------------------------
+# This file is part of solidity.
+#
+# solidity is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2019 solidity contributors.
+#------------------------------------------------------------------------------
+source test/externalTests/common.sh
+
+verify_input "$1"
+SOLJSON="$1"
+
+function gnosis_safe_test
+{
+    OPTIMIZER_LEVEL=1
+    setup https://github.com/gnosis/safe-contracts.git development
+    run_init "npm install"
+
+    CONFIG=$(find_truffle_config)
+    replace_libsolc_call
+
+    run_test "npx truffle compile" "npm test"
```

Comment:
This does not propagate errors correctly. I will need to rollback to a previous solution:
```
funcition init_fn { npm install; }
funcition compile_fn { npx truffle compile; }
funcition test_fn { npm test; }

run_init init_fn
...
run_test compile_fn test_fn
```